hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -217,8 +222,10 @@ public HoodieBackedTableMetadata metadata() {
    * Initialize the metadata table if it does not exist.
    *
    * If the metadata table did not exist, then file and partition listing is used to bootstrap the table.
+   * @param instantInProgressTimestamp Timestap of an instant in progress on the dataset. This instant is ignored
+   *                                   while deciding to bootstrap the metadata table.
    */
-  protected abstract void initialize(HoodieEngineContext engineContext);
+  protected abstract void initialize(HoodieEngineContext engineContext, Option<String> instantInProgressTimestamp);
```

Comment:
minor: may be we can rename to inflightInstantTimestamp.

Reply:
sure. renamed.